netty/src/main/java/io/grpc/netty/NettyServerStream.java

```diff
@@ -96,14 +96,18 @@ private class Sink implements AbstractServerStream.Sink {
     @Override
     public void writeHeaders(Metadata headers, boolean flush) {
       try (TaskCloseable ignore = PerfMark.traceTask("NettyServerStream$Sink.writeHeaders")) {
-        writeQueue.enqueue(
-            SendResponseHeadersCommand.createHeaders(
-                transportState(),
-                Utils.convertServerHeaders(headers)),
-            flush);
+        writeHeadersInternal(headers, flush);
       }
     }
 
+    private void writeHeadersInternal(Metadata headers, boolean flush) {
```

Comment:
Could we just do this inline?

We have the `*Internal` methods because 1) some methods were pretty large and when adding the PerfMark tracing we didn't want to change the entire method's indentation and 2) some methods had lots of indentation already.

Reply:
Done. I've inlined writeFrameInternal as well - it ended up to be small too.